source/common/config/type_to_endpoint.cc

```diff
@@ -9,9 +9,22 @@
 #include "envoy/api/v2/srds.pb.h"
 #include "envoy/service/discovery/v2/rtds.pb.h"
 
+#include "envoy/api/v3alpha/auth/cert.pb.h"
+#include "envoy/api/v3alpha/cds.pb.h"
+#include "envoy/api/v3alpha/eds.pb.h"
+#include "envoy/api/v3alpha/lds.pb.h"
+#include "envoy/api/v3alpha/rds.pb.h"
+#include "envoy/api/v3alpha/route/route.pb.h"
+#include "envoy/service/route/v3alpha/srds.pb.h"
+#include "envoy/service/discovery/v3alpha/rtds.pb.h"
+
+#include "common/common/assert.h"
 #include "common/grpc/common.h"
 
 // API_NO_BOOST_FILE
+// TODO(htuch): the cross product of transport, resource version and type is
```

Comment:
Looking at this file, I think we need someting like #9454 to make this sane in the long term (or even now).

Reply:
@lizan I'll experiment with this tomorrow. So far, I'm thinking:
1. Introduce an annotation on each xDS RPC method that points back to the resource type. This annotation will live in Envoy, rather than UDPA, for now, since we haven't finalized how UDPA-TP looks like.
2. When Envoy starts, we scan the entire descriptor pool, and build a reverse mapping from resource type to RPC methods for SoTW, delta gRPC and REST.
3. `typed_to_endpoint.cc` performs a lookup in this reverse map using the existing interface.
LMK if you have different ideas on the design, I think this could be hacked out fairly quickly once we settle on this.